contracts/jobs/libraries/JobLib.sol

```diff
@@ -104,7 +104,7 @@ library JobLib {
      * @param _segmentSequenceNumber Segment sequence number in stream
      * @param _dataHash Content-addrssed storage hash of segment data
      */
-    function personalSegmentHash(string _streamId, uint256 _segmentNumber, bytes32 _dataHash) public constant returns (bytes32) {
+    function personalSegmentHash(string _streamId, uint256 _segmentNumber, bytes32 _dataHash) public pure returns (bytes32) {
```

Comment:
What makes these two functions candidates for `pure` instead of `view`? This is just a learning moment for those following along.

Reply:
`pure` functions do not read from or write to contract state. `view` functions can read from, but cannot write to contract state. Right now these declarations just allow the compiler to check if you are accidentally violating the associated restrictions for a `pure`/`view` declaration, but in the future the compiled bytecode for `view` functions might use the new STATICCALL opcode which actually enforces read-only calls during runtime on the EVM level